src/components/input/input.scss

```diff
@@ -320,11 +334,16 @@ md-input-container.md-icon-float {
 
 md-input-container.md-icon-right {
   input {
-    margin-right: $icon-offset;
+    @include rtl(margin-right, $icon-offset, initial);
+    @include rtl(margin-left, initial, $icon-offset);
     + md-icon {
       top: 26px;
-      right: 2px;
-      margin-right: 0;
+
+      @include rtl(right, 2px, initial);
+      @include rtl(left, initial, 2px);
+
+      @include rtl(margin-right, 0, initial);
+      @include rtl(margin-left, initial, 0);
```

Comment:
In this case, we might not be able to use the `rtl` helper method as it stands.  What we essentially want is to ONLY define `right` or `left` conditionally - otherwise, just not assign any value.